### PR TITLE
bug: fix upstream url

### PR DESCRIPTION
**Pull request by michaelmerrill**
URL is not defined in the types file for the upstream property. You may want this to be a string, but by import URL it fixed the error.

I also updated the documentation to include the 'upstream' property.

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ import { ReaderTaskEither } from 'fp-ts/ReaderTaskEither';
 import { TaskEither } from 'fp-ts/TaskEither';
 import { Logger } from 'pino';
 import { NonEmptyArray } from 'fp-ts/NonEmptyArray';
+import { URL } from 'url';
```
Comment (XVincentX) [inline packages/core/src/types.ts:8]: This is not required, are you receiving any error?

**Review by XVincentX — CHANGES_REQUESTED** [review]
Thanks for the documentation changes.

I've put a small comment, I guess I do not understand why it's required to import URL; can you show me the specific error you're receiving?

File: packages/core/src/types.ts (hunk 1 above)
Comment (michaelmerrill) [reply packages/core/src/types.ts:8]: You're right, it's not required. I guess the issue I was running into was that without the URL type in the module typescript seems to think that `upstream` is an `any` type and was not warning me when I passed in a string as a url.  Maybe this is an issue with my tsconfig or vscode?

Screen Shot 2020-09-09 at 4 55 52 PM
Screen Shot 2020-09-09 at 4 57 08 PM

Comment (XVincentX) [reply packages/core/src/types.ts:8]: I'll take a look into. Can you remote the import in meantime?

Comment (XVincentX) [reply packages/core/src/types.ts]: I've done that for you. This is good to merge.

**Review by XVincentX — APPROVED** [review]
Good to go, thanks for the change

**Pull request merged**